sormas-ui/src/main/java/de/symeda/sormas/ui/person/PersonSelectionField.java

```diff
@@ -65,11 +70,17 @@
 	 * @param infoText
 	 *            Information displayed to the user.
 	 */
-	public PersonSelectionField(PersonDto referencePerson, String infoText) {
+	public PersonSelectionField(PersonDto referencePerson, String infoText, String infoTextWithoutMatches) {
 		this.referencePerson = referencePerson;
-		this.infoText = infoText;
 
 		initializeGrid();
+
+		this.hasMatches = this.hasMatches();
+		if (infoTextWithoutMatches == null) {
+			this.infoText = infoText;
+		} else {
+			this.infoText = hasMatches ? infoText : infoTextWithoutMatches;
+		}
```

Comment:
I find the logic quite disadvantegous:
1. It invokes `this.hasMatches()` in scenarios where it was not called before the change and calling it probably still is not necessary. That is bad because it's an expensive call. One could circumvent this like that:

```suggestion
		if (infoTextWithoutMatches == null) {
			this.infoText = infoText;
		} else {
			this.infoText = this.hasMatches() ? infoText : infoTextWithoutMatches;
		}
```

2. The personSelectionField is mostly managed from outside (e.g. whether the field is shown or not is decided outside of this class). This logic makes a decision inside the class now.